File: dart/triangle/lib/triangle.dart
// Determine if a triangle is equilateral, isosceles or scalene.
// https://exercism.org/tracks/dart/exercises/triangle
class Triangle {
  bool equilateral(double a, double b, double c) {
    return a == b && a == c;
  }

  bool isosceles(double a, double b, double c) {
    return (a == b && a != c) || (b == c && b != a) || (a == c && a != b);
  }

  bool scalene(double a, double b, double c) {
    return a != b && a != c && b != c;
  }
}

File: ruby/triangle/triangle.rb
# frozen_string_literal: false

# https://exercism.org/tracks/ruby/exercises/triangle
# Triangle exercise
class Triangle
  def initialize(sides)
    @side_a = sides[0]
    @side_b = sides[1]
    @side_c = sides[2]
  end

  def triangle?
    return false if @side_a.zero? || @side_b.zero? || @side_c.zero?

    @side_a + @side_b > @side_c && @side_a + @side_c > @side_b && @side_b + @side_c > @side_a
  end

  def equilateral?
    triangle? && @side_a == @side_b && @side_a == @side_c
  end

  def isosceles?
    triangle? && (
      @side_a == @side_b ||
      @side_b == @side_c ||
      @side_a == @side_c
    )
  end

  def scalene?
    triangle? && @side_a != @side_b && @side_a != @side_c && @side_b != @side_c
  end
end
